packages/twenty-front/src/modules/object-record/record-board/hooks/useRecordBoardSelection.ts

```diff
@@ -2,13 +2,25 @@ import { useRecoilCallback } from 'recoil';
 
 import { getActionMenuDropdownIdFromActionMenuId } from '@/action-menu/utils/getActionMenuDropdownIdFromActionMenuId';
 import { getActionMenuIdFromRecordIndexId } from '@/action-menu/utils/getActionMenuIdFromRecordIndexId';
-import { useRecordBoardStates } from '@/object-record/record-board/hooks/internal/useRecordBoardStates';
+import { isRecordBoardCardSelectedComponentFamilyState } from '@/object-record/record-board/states/isRecordBoardCardSelectedComponentFamilyState';
+import { recordBoardSelectedRecordIdsComponentSelector } from '@/object-record/record-board/states/selectors/recordBoardSelectedRecordIdsComponentSelector';
 import { isDropdownOpenComponentState } from '@/ui/layout/dropdown/states/isDropdownOpenComponentState';
+import { useRecoilComponentCallbackStateV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentCallbackStateV2';
 import { extractComponentState } from '@/ui/utilities/state/component-state/utils/extractComponentState';
+import { getSnapshotValue } from '@/ui/utilities/state/utils/getSnapshotValue';
 
 export const useRecordBoardSelection = (recordBoardId: string) => {
-  const { selectedRecordIdsSelector, isRecordBoardCardSelectedFamilyState } =
-    useRecordBoardStates(recordBoardId);
+  const isRecordBoardCardSelectedFamilyState =
+    useRecoilComponentCallbackStateV2(
+      isRecordBoardCardSelectedComponentFamilyState,
+      recordBoardId,
+    );
+
+  const recordBoardSelectedRecordIdsSelector =
```

Comment:
I think we should also extract selection at recordIndex level too (but that's another story!)

Reply:
Yes I think too, I didn't make it here because the PR was big enough 😅